app/src/main/java/pm/gnosis/heimdall/ui/addressbook/edit/AddressBookEditEntryActivity.kt

```diff
@@ -0,0 +1,127 @@
+package pm.gnosis.heimdall.ui.addressbook.edit
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import com.jakewharton.rxbinding2.view.clicks
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.rxkotlin.plusAssign
+import io.reactivex.rxkotlin.subscribeBy
+import kotlinx.android.synthetic.main.layout_address_book_update_entry.*
+import pm.gnosis.heimdall.R
+import pm.gnosis.heimdall.di.components.ViewComponent
+import pm.gnosis.heimdall.helpers.ToolbarHelper
+import pm.gnosis.heimdall.reporting.ScreenId
+import pm.gnosis.heimdall.ui.addressbook.AddressBookContract
+import pm.gnosis.heimdall.ui.addressbook.list.AddressBookActivity
+import pm.gnosis.heimdall.ui.base.ViewModelActivity
+import pm.gnosis.heimdall.ui.qrscan.QRCodeScanActivity
+import pm.gnosis.heimdall.utils.handleQrCodeActivityResult
+import pm.gnosis.heimdall.utils.parseEthereumAddress
+import pm.gnosis.model.Solidity
+import pm.gnosis.models.AddressBookEntry
+import pm.gnosis.svalinn.common.utils.snackbar
+import pm.gnosis.svalinn.common.utils.subscribeForResult
+import pm.gnosis.svalinn.common.utils.toast
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.asEthereumAddressString
+import pm.gnosis.utils.exceptions.InvalidAddressException
+import timber.log.Timber
+import javax.inject.Inject
+
+class AddressBookEditEntryActivity : ViewModelActivity<AddressBookContract>() {
+    @Inject
+    lateinit var toolbarHelper: ToolbarHelper
+
+    override fun layout() = R.layout.layout_address_book_update_entry
+
+    override fun inject(component: ViewComponent) = component.inject(this)
+
+    override fun screenId() = ScreenId.ADDRESS_BOOK_EDIT_ENTRY
+
+    private lateinit var address: Solidity.Address
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        intent.getStringExtra(EXTRA_ADDRESS).let {
+            it.asEthereumAddress()?.let {
+                address = it
+            } ?: run {
+                toast(R.string.invalid_ethereum_address)
+                finish()
+                return
+            }
+        }
+
+        // Needs to be done on onCreate because the result of the QR Scan activity would get "overridden" by the DB load
```

Comment:
Again we should define if that should really be possible.

Reply:
Also just because you trigger this in "onCreate" doesn't mean that it finishes before the `onActivityResult`, so you are introducing a race condition, we should change this.